File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        l = 0
        r = n - 1

        for i in range(len(nums)-1):
            l = i+1
            r = len(nums)-1
            t = target - nums[i]
            while l <= r:
                m = (l+r)//2

                if nums[m] > t:
                    r = m-1
                elif nums[m] < t:
                    l = m+1
                else:
                    return [i+1,m+1]
        return [-1,-1]


        
        

File: Data Structures & Algorithms/search-2d-matrix/submission-0.java
class Solution {

    public int[] coord(int mid, int[][] matrix){
        int n = matrix[0].length;
        return new int[]{mid/n,mid%n};
    }


    public boolean searchMatrix(int[][] matrix, int target) {
        

        int l = 0;
        int r = matrix.length * matrix[0].length - 1;

        int m = 0;
        
        while(l <= r){
            m = (l+r)/2;
            int[] ll = new int[2]; 
            ll = coord(m,matrix);
            int c = matrix[ll[0]][ll[1]]; 
            //System.out.println(c);
            if(c == target){return true;}
            else if(c < target){l = m+1;}
            else{r = m-1;}
        }
        return false;

    }
}
